src/main/java/eod/warObject/leader/red/Sundar.java

```diff
@@ -1,29 +1,136 @@
+        canHandle.add(ObjectDeadEvent.class);
+    }
+
+    public void deathPulse() {
+        ArrayList<Point> targets = getEightWays();
+        attack(targets, 4);
     }
 
     @Override
-    protected ArrayList<NormalCard> generateSpecialOrder() {
-        ArrayList<NormalCard> deck = new ArrayList<>();
+    public void attack(ArrayList<Point> targets, int hp) {
+        this.damage(hp);
+        Gameboard gameboard = player.getBoard();
+        for(Point p:targets) {
+            try {
+                WarObject target = gameboard.getObjectOn(p.x, p.y);
+                if(target instanceof Bunker || target instanceof Machine) {
+                    continue;
+                }
+                if(target.getPlayer().equals(player)) {
+                    if(target instanceof Ghost) {
+                        ((Ghost) target).attack();
+                    }
+                } else {
+                    ((Damageable) target).damage(hp);
+                }
+            } catch (Exception e) {}
+        }
+    }
+
+    public ArrayList<Point> getEightWays() {
+        ArrayList<Point> points = new ArrayList<>();
+        int boardSize = Gameboard.boardSize;
+        int x = position.x-1, y = position.y;
+        while (x >= 0) {
+            points.add(new Point(x, y));
+            x--;
+        }
+        x = position.x+1;
+        while (x < boardSize) {
+            points.add(new Point(x, y));
+            x++;
+        }
+        x = position.x;
+        y = position.y-1;
+        while (y >= 0) {
+            points.add(new Point(x, y));
+            y--;
+        }
+        y = position.y+1;
+        while (y < boardSize) {
+            points.add(new Point(x, y));
+            y++;
+        }
+        x = position.x-1;
+        y = position.y-1;
+        while (x >= 0 && y >= 0) {
+            points.add(new Point(x, y));
+            x--;
+            y--;
+        }
+        x = position.x+1;
+        y = position.y+1;
+        while (x < boardSize && y < boardSize) {
+            points.add(new Point(x, y));
+            x++;
+            y++;
+        }
+        return points;
+    }
+
+    @Override
+    protected ArrayList<Card> generateSpecialOrder() {
+        ArrayList<Card> deck = new ArrayList<>();
         deck.add(new EquivalentExchange(player));
-        // TODO: add another special order
+        deck.add(new DeathPulse(player));
         return deck;
     }
+
+    @Override
+    public ArrayList<Class<? extends Event>> supportedEventTypes() {
+        return canHandle;
+    }
+
+    @Override
+    public void onEventOccurred(GameObject sender, Event event) {
+        if (event instanceof ObjectDeadEvent) {
+            Damageable deadObject = ((ObjectDeadEvent) event).getDeadObject();
+            WarObject object = (WarObject) deadObject;
+            int x = object.position.x, y = object.position.y;
+            if(deadObject instanceof Ghost) {
+                heal(2);
```

Comment:
Login problem.
According to card specification, the first passive skill is described as follow:
> 1.己方的「亡靈」死亡時，回復自己2點生命值
However, no ghost's owner check is found.